coralproject/talk
# recaptcha + RL support

## What does this PR do?

Adds recaptcha support for logging into the admin panel. Realistically, we're going to have to add this to the embed eventually since the logins are shared.

## How do I test this PR?

1. `yarn install`
2. start `ngrok` and point it at `localhost:3000` or whatever you're running it at
3. sign up [reCAPTCHA](https://www.google.com/recaptcha/admin#list)
4. add the ngrok url to your list of domains
5. set two environment variables `TALK_RECAPTCHA_SECRET` and `TALK_RECAPTCHA_PUBLIC` in your `.env` file.
6. fully restart the server and the fe build process (**IMPORTANT** you're going to have to start the build with `dotenv yarn run build-watch` from now on)
7. navigate to the ngrok url from before
8. try to log in wrong a few times to the admin
9. you should see a different error letting you know that you've reached the max # of failed attempts
10. you should see a recaptcha widget appear below the login form
11. update your information to the correct creds and complete the [I Am Not A Robot](https://www.youtube.com/watch?v=S_oMD6-6q5Y) thing
12. log in successfully
